src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/virtualusers/MulticastVirtualUsers.kt

```diff
@@ -52,16 +57,29 @@ class MulticastVirtualUsers<out T : VirtualUsers>(
             throw Exception("$virtualUsers virtual users are not enough to spread into $nodeCount nodes")
         }
         val loadSlices = load.slice(nodeCount)
+        val activeNodes = ConcurrentHashMap.newKeySet<String>()
```

Comment:
```suggestion
        val activeNodes = ConcurrentHashMap.newKeySet<T>()
```
And then we don't have `toString` later on. We can rely on `equals` and `hashCode` rather than hoping for unique `toString` values. All known `VirtualUsers` impls have identity equality. And I don't foresee a case, where different physical VU nodes should be considered equal, so no future impl should hurt us here.